### PR TITLE
Implemented Document Delivery

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#8366

*Description of changes:*
* Some Review view layout updates.
* Relocated Document delivery documents to its own section.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/assets/styles/base.scss [hunk 1]
```diff
@@ -74,6 +74,10 @@ h3 {
   line-height: 2.25rem;
 }
 
+label {
+  font-size: 1rem;
```
Comment (severinbeauvais) [inline src/assets/styles/base.scss:78]: Will this affect other labels ? How are you verifying that?

File: src/assets/styles/overrides.scss [hunk 2]
```diff
@@ -21,6 +21,15 @@
   transform: translateY(-8px) scale(0.75);
 }
 
+// V-Cards
+.v-card {
+  .v-card__text {
+    font-weight: 300;
```
Comment (severinbeauvais) [inline src/assets/styles/overrides.scss:27]: Typo? In base.scss we only declare two font weights (400 and 700).

(And personally, I'd prefer to see "regular" and "bold". I shudder at the thought of different bold levels :P )

File: src/components/ReviewConfirm/DocumentDelivery.vue [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<template>
+  <v-card flat class="pr-5 pb-4">
```
Comment (severinbeauvais) [inline src/components/ReviewConfirm/DocumentDelivery.vue:2]: It's better for this component's parent to specify the whitespace around it.

File: src/store/getters/state-getters.ts [hunk 4]
```diff
@@ -297,11 +297,5 @@ export const getCertifyState = (state: StateIF): CertifyIF => {
...
 /** The Completing Party's email address. */
 export const getCompletingPartyEmail = (state: StateIF): string => {
-  const completingParty =
-    getAddPeopleAndRoleStep(state).orgPeople.find(person => {
-      return Boolean(person.roles?.some(role => {
-        return (role.roleType === RoleTypes.COMPLETING_PARTY)
-      }))
-    })
-  return (completingParty?.officer?.email || null)
+  return (state.stateModel.tombstone.userEmail)
```
Comment (severinbeauvais) [inline src/store/getters/state-getters.ts:300]: Does this business logic change work for BEN IAs too?

Also, you can use `getTombstone(state).userEmail` now :)

**Review by severinbeauvais — COMMENTED** [review]
Looks good. Are there any unit tests?

File: src/assets/styles/base.scss (hunk 1 above)
Comment (cameron-freshworks) [reply src/assets/styles/base.scss:78]: Hopefully all of them 😁 
This UI was a little bit behind on our modern designs, so this should bring them into sync with what you would see in the Edit ui for example. 

I just verified visually and it updated everything i'd expect it too. Certainly a little more global and perhaps out of scope of this ticket, but to meet requirements on my specific ticket i figured it made sense to update them all now.

File: src/assets/styles/overrides.scss (hunk 2 above)
Comment (cameron-freshworks) [reply src/assets/styles/overrides.scss:27]: oo yea, this was a mistake. Will fix!

Comment (cameron-freshworks) [reply src/assets/styles/overrides.scss:27]: That being said, this is redundant. Removed!

File: src/components/ReviewConfirm/DocumentDelivery.vue (hunk 3 above)
Comment (cameron-freshworks) [reply src/components/ReviewConfirm/DocumentDelivery.vue:2]: kk, will update

Comment (cameron-freshworks) [reply src/components/ReviewConfirm/DocumentDelivery.vue:2]: Oh wait, this is padding within the component.

File: src/store/getters/state-getters.ts (hunk 4 above)
Comment (cameron-freshworks) [reply src/store/getters/state-getters.ts:300]: Yea! Works for both :)
Originally was only applying the email once the completing party had been filled out. 
Although we retrieved the email right up front and always had it.

**Comment by cameron-freshworks** [comment]
> Looks good. Are there any unit tests?

Nothing meaningful really 🤷  
I mean we could always add tests tho..

File: src/components/ReviewConfirm/DocumentDelivery.vue (hunk 3 above)
Comment (severinbeauvais) [reply src/components/ReviewConfirm/DocumentDelivery.vue:2]: Are you sure this isn't padding AROUND the component? Looks like the v-container on line 3 specifies padding within the component.

**Comment by severinbeauvais** [comment]
> > Looks good. Are there any unit tests?
> 
> Nothing meaningful really 🤷
> I mean we could always add tests tho..

How about a unit test file for your new component with some basic tests?

File: src/components/ReviewConfirm/DocumentDelivery.vue (hunk 3 above)
Comment (cameron-freshworks) [reply src/components/ReviewConfirm/DocumentDelivery.vue:2]: Ah yea, there is another level of padding inside in the v-container. Maybe i can do away with the container. 
The card itself is the component tho

**Comment by cameron-freshworks** [comment]
> > > Looks good. Are there any unit tests?
> > 
> > 
> > Nothing meaningful really 🤷
> > I mean we could always add tests tho..
> 
> How about a unit test file for your new component with some basic tests?

You got it!

File: src/components/ReviewConfirm/DocumentDelivery.vue (hunk 3 above)
Comment (cameron-freshworks) [reply src/components/ReviewConfirm/DocumentDelivery.vue:2]: Fixed up, in next commit!

**Comment by cameron-freshworks** [comment]
> > > > Looks good. Are there any unit tests?
> > > 
> > > 
> > > Nothing meaningful really 🤷
> > > I mean we could always add tests tho..
> > 
> > 
> > How about a unit test file for your new component with some basic tests?
> 
> You got it!

Updated 👍

File: tests/unit/DocumentDelivery.spec.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    },
+    defineCompanyStep: {
+      businessContact: {
+        email: 'BENBusinessContact@email.com'
+      }
+    }
+  }
+]
+
+for (const mock of documentDeliveryCases) {
```
Comment (severinbeauvais) [inline tests/unit/DocumentDelivery.spec.ts:30]: Fancy! And great! ⭐

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**